pkg/globalnet/controllers/global_ingressip_controller.go

```diff
@@ -148,12 +177,56 @@ func (c *globalIngressIPController) onDelete(ingressIP *submarinerv1.GlobalIngre
 
 	key, _ := cache.MetaNamespaceKeyFunc(ingressIP)
 
+	if ingressIP.Spec.ServiceRef != nil {
+		chainName := ingressIP.GetAnnotations()[kubeProxyIPTableChainAnnotation]
+		if chainName == "" {
+			klog.Warningf("%q annotation is missing on %q", kubeProxyIPTableChainAnnotation, key)
+			return nil, true
```

Comment:
Returning true to re-queue won't have any effect since the object has been deleted and the annotation wouldn't ever be added. Also returning here will elide releasing the IP on L194. Perhaps we should log the warning and skip `RemoveIngressRulesForService` but still proceed to L194.